prediction_market_agent/agents/ofvchallenger_agent/deploy.py

```diff
@@ -1,21 +1,146 @@
+
+        markets_open_for_answers = OmenSubgraphHandler().get_omen_binary_markets(
+            limit=None,
+            creator_in=MARKET_CREATORS_TO_CHALLENGE,
+            # We need markets already opened for answers.
+            opened_before=utcnow(),
+            # With a little bandwidth for the market to be finalized,
+            # so we have time for processing it without erroring out at the end.
+            finalized_after=utcnow() + timedelta(minutes=30),
+        )
+        logger.info(f"Found {len(markets_open_for_answers)} markets to challenge.")
+
+        for market in markets_open_for_answers:
+            self.challenge_market(market, api_keys)
+
+    @observe()
+    def challenge_market(
+        self,
+        market: OmenMarket,
+        api_keys: APIKeys,
+        web3: Web3 | None = None,
+    ) -> Challenge | None:
+        logger.info(f"Challenging market {market.url=}")
+
+        existing_responses = OmenSubgraphHandler().get_responses(
+            question_id=market.question.id
+        )
+        logger.info(
+            f"{market.url=}'s responses and bonds: {[(r.answer, r.bond_xdai) for r in existing_responses]}"
+        )
+
+        # Next bond needs to be at least double the previous one.
+        if any(
+            response.bond_xdai >= CHALLENGE_BOND / 2 for response in existing_responses
+        ):
+            logger.info(
+                f"Market {market.url=} already challenged with bond > {CHALLENGE_BOND} / 2. Skipping."
+            )
+            return None
+
+        # We don't plan to re-challenge markets already challenged by the challenger, should we?
+        if any(
+            response.user_checksummed == OFV_CHALLENGER_SAFE_ADDRESS
+            for response in existing_responses
+        ):
+            logger.info(
+                f"Market {market.url=} already challenged by replicator. Skipping."
+            )
+            return None
+
+        try:
+            answer = ofv_answer_binary_question(market.question_title, api_keys)
+        except Exception as e:
+            logger.exception(
+                f"Exception while getting factuality for market {market.url=}. Skipping. Exception: {e}"
+            )
+            return None
+
+        if answer is None or answer.factuality is None:
+            logger.error(
+                f"Failed to get factuality for market {market.url=}, question {market.question_title=}. Skipping."
+            )
+            return None
+
+        new_resolution = Resolution.from_bool(answer.factuality)
+        logger.info(
+            f"Challenging market {market.url=} with resolution {new_resolution=}"
+        )
+
+        omen_submit_answer_market_tx(
+            api_keys=api_keys,
+            market=market,
+            resolution=new_resolution,
+            bond=CHALLENGE_BOND,
+            web3=web3,
+        )
+
+        return Challenge(
+            old_responses=existing_responses, new_resolution=new_resolution
+        )
```

Comment:
I was expecting this section to only run if the new resolution disagreed with the existing one (else return None)

Reply:
> I was expecting this section to only run if the new resolution disagreed with the existing one (else return None).

Martin agreed to $10 bond, so we can challenge everything to increase the bond to attract people (even if OFV and Olas agree, it doesn't mean it's correct). 

> Also, looks like the returned Challenge object isn't used anywhere (so maybe not necessary?)

It will be saved in Langfuse, which is just a practicality for debugging.

Reply:
> Martin agreed to $10 bond and losing 10% on 15 markets daily, so we can challenge everything to increase the bond to attract people (even if OFV and Olas agree, it doesn't mean it's correct).

Ah okay, we're taking the nuclear option!